src/agent/agent.c

```diff
@@ -370,27 +370,27 @@ bool agent_set_port(Agent *agent, const char *port_s) {
         return true;
 }
 
-bool agent_set_host(Agent *agent, const char *host) {
-        char *dup = strdup(host);
+static bool agent_set_str(char **p, const char *s) {
```

Comment:
Since this is just copying the source to the target, how about we move it to `libhirte/common/common.c` (needs to be created) and rename it to `copy_string` or sth similar? Could simplify quite some code throughout hirte.

Reply:
Done